### PR TITLE
Feat/tasks event

### PR DESCRIPTION
**Pull request by MaxProtos**

### CODE REVIEW

File: frontend/src/app/libs/code-editors/monaco-editor/monaco-editor.component.html [hunk 1]
```diff
@@ -1 +1,5 @@
-<div class="editor-container" #editorContainer></div>
+<div class="editor-wrapper">
+  <div class="editor-container" #editorContainer>
```
Comment (maximwow70) [inline frontend/src/app/libs/code-editors/monaco-editor/monaco-editor.component.html:2]: Why we need additional wrapper here?

File: frontend/src/app/libs/code-editors/monaco-editor/monaco-editor.component.ts [hunk 2]
```diff
@@ -36,6 +45,10 @@ export class MonacoEditorComponent implements AfterViewInit, OnChanges, OnDestro
 
 	private destroy$: Subject<boolean> = new Subject<boolean>();
 
+  public onChange = (_: string) => {};
+
+  public onBlur = () => {};
```
Comment (maximwow70) [inline frontend/src/app/libs/code-editors/monaco-editor/monaco-editor.component.ts:50]: Type should be provided.

File: frontend/src/app/libs/common-components/components/select/select.component.ts [hunk 3]
```diff
@@ -15,37 +16,54 @@ import { FormControlCommonDirective } from '../../directives/formControl/form-co
...
+    const MAX_HOUR: number = 23;
+    const TEN_HOURS: number = 10;
+    const arrayTime: ISelectOption[] = [];
+
+    for (let i: number = 0; i <= MAX_HOUR; i++) {
+      let hour: string = i.toString();
+      if (i < TEN_HOURS) {
+        hour = '0' + hour;
+      }
+      arrayTime.push({
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/select/select.component.ts:57]: You can use spread operator here like:
```
arrayTime = [
   ...arrayTime,
   {...},
   {...}
]
```

File: frontend/src/app/libs/common-components/components/slide-toggle/slide-toggle.component.ts [hunk 4]
```diff
@@ -30,6 +30,9 @@ export class SlideToggleComponent extends FormControlCommonDirective {
 	@Input()
 	public value: boolean;
 
+	@Input()
+  public labelPosition: 'before' | 'after';
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/slide-toggle/slide-toggle.component.ts:34]: Please, define new type for labelPosition.
Just `LabelPosition`.

File: frontend/src/app/libs/common-components/components/tab-navigation/tab-navigation.component.ts [hunk 5]
```diff
@@ -1,32 +1,58 @@
...
 	}
 
 	public get invitationsTab(): boolean {
-		return this.currentTab === 'invitationsTab';
+		return this.currentTab === Tabs.invitationsTab;
 	}
 
+  public ngOnInit(): void {
+    this.eventFacadeService.tasks$
+    .pipe(
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/tab-navigation/tab-navigation.component.ts:48]: Looks like here something happened with formatting.

File: frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.html [hunk 6]
```diff
@@ -0,0 +1,160 @@
+<form [formGroup]="taskCreatorControl" class="tasks-creator">
+  <ng-container *ngIf="!isListEmpty; else emptyList">
+    <div class="tasks-creator__header event-gradient">
+
+      <div *ngIf="!isTaskNameEditMode" class="tasks-creator__name-editor">
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.html:5]: class always should be the first attribute of element to good readability.

File: frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.ts [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+    super();
+    this.taskCreatorControl = this.formBuilder.group({
+      selectedTaskTypeControl: new FormControl(),
+      maxScoreControl: new FormControl(),
+      mentorCheckControl: new FormControl(),
+      textEditorControl: new FormControl(),
+    });
+
+    this.translateService.get([this.langTaskTypesKey])
+    .pipe(takeUntil(this.destroyedSource$))
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.ts:60]: The same with formatting.

File: frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.ts [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+    .subscribe(
+      (labels: object) => {
+        this.taskTypeOptions.length = 0;
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.ts:63]: Looks like a hack.
Please, use map here and reassign taskTypeOptions.

File: frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.ts [hunk 9]
```diff
@@ -0,0 +1,352 @@
...
+    this.taskCreatorControl.removeControl('taskNameControl');
+    this.isTaskNameEditMode = false;
+  }
+
+  public ngOnInit(): void {
+    this.rebuildFormControls(this.selectedTaskType);
+
+    this.taskCreatorControl.valueChanges.subscribe(
+      () => {
+        if (this.selectedTaskType === TasksTypes.coding && this.isMentorCheckSelected === false) {
```
Comment (maximwow70) [inline frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.ts:177]: `this.isMentorCheckSelected === false` the same as `!this.isMentorCheckSelected`

File: frontend/src/app/libs/event-creation/event-creation.component.ts [hunk 10]
```diff
@@ -1,56 +1,63 @@
...
+
+      default:
+    }
+  }
 
 	public getFormFromGeneralComponent(data: FormGroup): void {
 		this.formFromGeneralComponent = data;
 	}
+
+  public getTaskCreationForm(task: ITask): void {
```
Comment (maximwow70) [inline frontend/src/app/libs/event-creation/event-creation.component.ts:56]: Function name started with get, then this function return something.
Looks like here need to name set..

File: frontend/src/app/libs/event-creation/store/event.actions.ts [hunk 11]
```diff
@@ -0,0 +1,84 @@
+import { Action } from '@ngrx/store';
+import { GeneralEvent } from './model/model';
+import { ITask } from '../../common-components/components/tasks-creator/models/task';
+
+export enum EventsActionTypes {
+  LoadTasks = '[NewEvent] get tasks from backend',
+  LoadTasksSuccess = '[NewEvent] get tasks from backend successfully',
+  LoadTasksFail = '[NewEvent] get tasks from backend failed',
+	SetTitleForGeneralEvent = '[NewEvent] set title for new event',
```
Comment (maximwow70) [inline frontend/src/app/libs/event-creation/store/event.actions.ts:9]: Something happened with formatting.

File: frontend/src/app/libs/event-creation/store/event.facade.ts [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+import { ITask } from '../../common-components/components/tasks-creator/models/task';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class NewEventFacadeService {
+
+	constructor(private store: Store<GeneralEventState>) { }
+
+	get generalTitle$(): Observable<string> {
```
Comment (maximwow70) [inline frontend/src/app/libs/event-creation/store/event.facade.ts:23]: public missing.

File: frontend/src/app/libs/event-creation/store/event.facade.ts [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+	  return this.store.pipe(select(selectTasksEvent));
+  }
+
+	public setTitleForNewEvent(title: string): void {
+		this.store.dispatch(new SetTitleAction(title));
+	}
+
+	public loadTasks(): void {
+	  this.store.dispatch(new LoadTasksEventAction());
+  }
```
Comment (maximwow70) [inline frontend/src/app/libs/event-creation/store/event.facade.ts:37]: Formatting.

File: frontend/src/app/libs/event-creation/store/event.reducer.ts [hunk 14]
```diff
@@ -0,0 +1,149 @@
...
+          ...state.tasks,
+          error: action.payload as Error,
+        }
+      };
+    }
+
+		case EventsActionTypes.CreateGeneralEventFail: {
+			return {
+				...state,
+        general: {
```
Comment (maximwow70) [inline frontend/src/app/libs/event-creation/store/event.reducer.ts:136]: Please, fix formatting.

File: frontend/src/app/libs/common-components/components/button/button.component.ts [hunk 15]
```diff
@@ -31,6 +31,9 @@ export class ButtonComponent {
 	@Input()
 	public disableHover: boolean = false;
 
+	@Input()
+  public iconButton: boolean = false;
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/button/button.component.ts:35]: not sure about this input, we have icon-button to work with icons

File: frontend/src/app/libs/common-components/components/icon/icon.component.html [hunk 16]
```diff
@@ -1,5 +1,6 @@
 <mat-icon
 	*ngIf="icon"
-	class="icon icon--{{size}}">
+	class="icon icon--{{size}}"
+  [ngStyle]="{ 'color' : color }">
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/icon/icon.component.html:4]: [style.color]="color"

File: frontend/src/app/libs/common-components/components/input/input.component.html [hunk 17]
```diff
@@ -1,6 +1,7 @@
 <mat-form-field
 	appearance="outline"
 	[style.width] = "widthInput"
+  [ngClass]="{ 'no-paddings' : noPaddings }"
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/input/input.component.html:4]: same here

File: frontend/src/app/libs/common-components/components/input/input.component.html [hunk 18]
```diff
@@ -1,6 +1,7 @@
 <mat-form-field
 	appearance="outline"
 	[style.width] = "widthInput"
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/input/input.component.html:3]: unnecessary space near =

File: frontend/src/app/libs/common-components/components/select/models/select-option.ts [hunk 19]
```diff
@@ -0,0 +1,4 @@
+export interface ISelectOption {
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/select/models/select-option.ts:1]: just `SelectOption`

File: frontend/src/app/libs/common-components/components/slide-toggle/slide-toggle.component.ts (hunk 4 above)
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/slide-toggle/slide-toggle.component.ts:34]: create enum

File: frontend/src/app/libs/common-components/components/tab-navigation/models/tabs.ts [hunk 20]
```diff
@@ -0,0 +1,5 @@
+export enum Tabs {
+  'generalTab',
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/tab-navigation/models/tabs.ts:2]: set string values to better readability in templates, `Tabs { General: 'general' ... }`

File: frontend/src/app/libs/common-components/components/tab-navigation/tab-navigation.component.html [hunk 21]
```diff
@@ -1,6 +1,6 @@
 <div class="app-tab-navigation__block" [class.app-tab-navigation__block--active]="generalTab"
-	 (click)="changePage('generalTab')">GENERAL</div>
+	 (click)="changePage(tabsEnum.generalTab)">GENERAL</div>
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/tab-navigation/tab-navigation.component.html:2]: changePage('general')
or create 3 methods like `switchToGeneral()` and etc

File: frontend/src/app/libs/common-components/components/tab-navigation/tab-navigation.component.ts [hunk 22]
```diff
@@ -1,32 +1,58 @@
...
-		return this.currentTab === 'generalTab';
+		return this.currentTab === Tabs.generalTab;
 	}
 
 	public get tasksTab(): boolean {
-		return this.currentTab === 'tasksTab';
+		return this.currentTab === Tabs.tasksTab;
 	}
 
 	public get invitationsTab(): boolean {
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/tab-navigation/tab-navigation.component.ts:42]: `isInvitationsLabActive` or something like that

File: frontend/src/app/libs/common-components/components/tasks-creator/models/task.ts [hunk 23]
```diff
@@ -0,0 +1,18 @@
+import {TasksTypes} from './tasks-creator';
+
+export interface IAnswer {
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/tasks-creator/models/task.ts:3]: No need to use I for iterfaces, it's likely more .net practice :)

File: frontend/src/app/libs/common-components/components/tasks-creator/models/tasks-creator.ts [hunk 24]
```diff
@@ -0,0 +1,6 @@
+export enum TasksTypes {
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/tasks-creator/models/tasks-creator.ts:1]: Coding  = 'CODING'

File: frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.html [hunk 25]
```diff
@@ -0,0 +1,160 @@
...
+          [value]="taskTypesEnum.single"
+          (selectionChanged)="rebuildFormControls($event.value)"
+          formControlName="selectedTaskTypeControl"
+          [options]="taskTypeOptions"
+        ></app-select>
+
+        <div class="tasks-creator__mentor-score-container">
+          <app-slide-toggle
+            class="tasks-creator__mentor-check"
+            [labelPosition]="'before'"
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.html:63]: avoid cases like thin across application [...]="'...'" can be transformed to ...="..."

File: frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.html [hunk 26]
```diff
@@ -0,0 +1,160 @@
...
+      <ng-container [ngSwitch]="getSelectedTaskType()">
+
+        <ng-container *ngSwitchCase="taskTypesEnum.single">
+
+          <mat-radio-group formControlName="correctSingleAnswerControl">
+            <div formArrayName="answersArrayControl">
+              <div class="tasks-creator__radio-answer"
+                   *ngFor="let answer of singleAnswersControl.controls; let i = index"
+                   [formGroupName]="i">
+                <mat-radio-button class="mat-form-field-wrapper" value="{{ i.toString() }}"></mat-radio-button>
```
Comment (clain-atomi) [inline frontend/src/app/libs/common-components/components/tasks-creator/tasks-creator/tasks-creator.component.html:93]: no need .toString here

**Review by clain-atomi — CHANGES_REQUESTED** [review]

**Review by clain-atomi — CHANGES_REQUESTED** [review]

**Pull request merged**